librz/core/cmd/cmd_cmp.c

```diff
@@ -473,384 +103,324 @@ static void __core_cmp_bits(RzCore *core, ut64 addr) {
-			case 8: sz = '1'; break;
-			case 16: sz = '2'; break;
-			case 32: sz = '4'; break;
-			case 64: sz = '8'; break;
-			default: sz = '4'; break; // default
-			}
-		} else if (sz == '?') {
-			eprintf("Usage: cV[1248] [addr] @ addr2\n"
-				"Compare n bytes from one address to current one and return in $? 0 or 1\n");
-		}
-		sz -= '0';
-		if (sz > 0) {
-			ut64 at = rz_num_math(core->num, input + 2);
-			ut8 buf[8] = { 0 };
-			rz_io_read_at(core->io, at, buf, sizeof(buf));
-			core->num->value = memcmp(buf, core->block, sz) ? 1 : 0;
-		}
-		break;
-	}
-	case 'l': // "cl"
-		if (strchr(input, 'f')) {
-			rz_cons_flush();
-		} else if (input[1] == 0) {
-			rz_cons_fill_line();
-			// rz_cons_clear_line (0);
-		} else if (!strchr(input, '0')) {
-			rz_cons_clear00();
+// cb
+RZ_IPI RzCmdStatus rz_cmd_cmp_bytes_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	RzCmdStatus ret = RZ_CMD_STATUS_ERROR;
+	ut64 sz = rz_num_math(core->num, argv[2]);
+	if (sz > 8) {
+		RZ_LOG_ERROR("Cannot compare more than 8 bytes. Use the c command instead.\n");
+		return ret;
+	}
+
+	ut64 num = rz_num_math(core->num, argv[1]);
+	ut64 mask = -1;
+	mask >>= 8 - sz;
+	ut64 valid_num = num & mask;
+	RzCompareData *cmp = rz_core_cmp_mem_data(core, core->offset, (ut8 *)&valid_num, sz);
+	if (!cmp) {
+		goto end;
+	}
+	if (cmp->same) {
+		core->num->value = 0;
+	} else {
+		core->num->value = 1;
+	}
+
+	int val = rz_core_cmp_print(core, cmp, state);
+	if (val != -1) {
+		ret = RZ_CMD_STATUS_OK;
+	}
+
+end:
+	rz_core_cmp_free(cmp);
+	return ret;
+}
+
+// cc
+RZ_IPI RzCmdStatus rz_cmd_cmp_hex_block_handler(RzCore *core, int argc, const char **argv) {
+	ut64 addr = rz_num_math(core->num, argv[1]);
+	bool col = core->cons->columns > 123;
+	ut8 *b = malloc(core->blocksize);
+	if (b) {
+		memset(b, 0xff, core->blocksize);
+		rz_io_nread_at(core->io, addr, b, core->blocksize);
+		rz_print_hexdiff(core->print, core->offset, core->block, addr, b, core->blocksize, col);
+	}
+	free(b);
+	return RZ_CMD_STATUS_OK;
+}
+
+// ccc
+RZ_IPI RzCmdStatus rz_cmd_cmp_hex_diff_lines_handler(RzCore *core, int argc, const char **argv) {
+	ut32 oflags = core->print->flags;
+	core->print->flags |= RZ_PRINT_FLAGS_DIFFOUT;
+	ut64 addr = rz_num_math(core->num, argv[1]);
+	bool col = core->cons->columns > 123;
+	ut8 *b = malloc(core->blocksize);
+	if (b) {
+		memset(b, 0xff, core->blocksize);
+		rz_io_nread_at(core->io, addr, b, core->blocksize);
+		rz_print_hexdiff(core->print, core->offset, core->block, addr, b, core->blocksize, col);
+	}
+	free(b);
+	core->print->flags = oflags;
+	return RZ_CMD_STATUS_OK;
+}
+
+// ccd
+RZ_IPI RzCmdStatus rz_cmd_cmp_disasm_handler(RzCore *core, int argc, const char **argv) {
+	RzList *cmp = rz_core_cmp_disasm(core, core->offset, rz_num_math(core->num, argv[1]), core->blocksize);
+	bool ret = rz_core_cmp_disasm_print(core, cmp, false);
+	rz_list_free(cmp);
+	return ret ? RZ_CMD_STATUS_OK : RZ_CMD_STATUS_ERROR;
+}
+
+// cf
```

Comment:
i'm not convinced we should really add these comments... These can become outdated without anyone noticing and grepping for "// cf" is meh. Easier and more accurate to use `./sys/rizin_which.py cf` IMHO.

Reply:
Makes sense. Had totally forgotten about the `rizin_which.py` script.

Reply:
On a second thought, I think I should let them be there. It is makes it much easier to figure out which handler corresponds to what. Having a 3 letter comment won't hurt in my opinion. So if it isn't causing any problems, I think I should let it stay.

Reply:
> Having a 3 letter comment won't hurt in my opinion. So if it isn't causing any problems, I think it should stay.

I agree it doesn't hurt, right now, but I fear for the long-term maintainability of such comments and the fact that relying on them is not something one should do, because by relying on them you might find `cf` command but not `wz` (which did not have that comment on top) or many other commands that we converted but did not have any comment on top like this.

Either a solution is trustworthy for all cases, or it isn't IMO.

cc @thestr4ng3r @XVilka @wargio for your input as well.

Reply:
I don't have any strong opinion on that. Keeping them is fine, removing as well.